tesseract_collision/CMakeLists.txt

```diff
@@ -82,292 +59,54 @@ set(COVERAGE_EXCLUDE
-                                                  "$<INSTALL_INTERFACE:include>")
-
-set(PACKAGE_LIBRARIES
-    ${PROJECT_NAME}_core
-    ${PROJECT_NAME}_bullet
-    ${PROJECT_NAME}_fcl
-    ${PROJECT_NAME}_test_suite
-    ${PROJECT_NAME}_test_suite_benchmarks
-    ${PROJECT_NAME}_profile
-    ${PROJECT_NAME}_vhacd
-    ${PROJECT_NAME}_convex_decomposition)
-
-if(NOT MSVC)
-  # Create target for creating convex hull's from meshes
-  add_executable(create_convex_hull src/create_convex_hull.cpp)
-  target_link_libraries(
-    create_convex_hull
-    PUBLIC ${PROJECT_NAME}_core
-           ${BULLET_LIBRARIES}
-           Boost::boost
-           Boost::program_options
-           Eigen3::Eigen
-           tesseract::tesseract_common
-           tesseract::tesseract_geometry
-           console_bridge::console_bridge
-           octomap
-           octomath)
-  target_compile_options(create_convex_hull PRIVATE ${TESSERACT_COMPILE_OPTIONS_PRIVATE}
-                                                    ${TESSERACT_COMPILE_OPTIONS_PUBLIC})
-  target_compile_definitions(create_convex_hull PRIVATE ${TESSERACT_COMPILE_DEFINITIONS} ${BULLET_DEFINITIONS})
-  target_clang_tidy(create_convex_hull ENABLE ${TESSERACT_ENABLE_CLANG_TIDY})
-  target_cxx_version(create_convex_hull PRIVATE VERSION ${TESSERACT_CXX_VERSION})
-  target_include_directories(create_convex_hull PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>")
-  target_include_directories(create_convex_hull SYSTEM PUBLIC "${BULLET_ROOT_DIR}/${BULLET_INCLUDE_DIRS}")
-
-  list(APPEND PACKAGE_LIBRARIES create_convex_hull)
+# VHACD
+option(TESSERACT_BUILD_VHACD "Build VHACD components" ON)
+if(TESSERACT_BUILD_VHACD)
+  message("Building VHACD components")
+  add_subdirectory(vhacd)
 endif()
 
-configure_package(NAMESPACE tesseract TARGETS ${PACKAGE_LIBRARIES})
-
-# Mark cpp header files for installation
-install(
-  DIRECTORY include/${PROJECT_NAME}
-  DESTINATION include
-  FILES_MATCHING
-  PATTERN "*.h"
-  PATTERN "*.hpp"
-  PATTERN "*.inl"
-  PATTERN ".svn" EXCLUDE)
-
-# Add Examples
-add_subdirectory(examples)
-
-if(TESSERACT_ENABLE_TESTING)
+# Testing
+if(TESSERACT_ENABLE_TESTING AND TESSERACT_BUILD_BULLET AND TESSERACT_BUILD_FCL)
   message(AUTHOR_WARNING "COLLISION CMAKE VERSION: ${CMAKE_VERSION}")
   enable_testing()
   add_run_tests_target(ENABLE ${TESSERACT_ENABLE_RUN_TESTING})
   add_subdirectory(test)
 endif()
 
-if(TESSERACT_ENABLE_BENCHMARKING)
+# Benchmarks
+if(TESSERACT_ENABLE_BENCHMARKING AND TESSERACT_BUILD_BULLET AND TESSERACT_BUILD_FCL)
   add_subdirectory(test/benchmarks)
 endif()
 
+# Examples
+if(TESSERACT_BUILD_BULLET AND TESSERACT_BUILD_FCL)
+  add_subdirectory(examples)
+endif()
+
+# Package configuration
+install(EXPORT ${PROJECT_NAME}-targets NAMESPACE "tesseract::" DESTINATION lib/cmake/${PROJECT_NAME})
+export(EXPORT ${PROJECT_NAME}-targets NAMESPACE "tesseract::"
+       FILE ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}-targets.cmake)
+configure_package()
```

Comment:
This can now be replaced with `configure_pacakge(NAMESPACE tesseract)`.

Reply:
Does this require `ros-industrial-cmake-boilerplate` > 0.2.11? On my machine the `tesseract_collision-targets` file does not get installed without explicitly doing it

Reply:
It requires master branch.

Reply:
Can you push a tag 2.12.0 for that change? I'll update the `dependencies.rosinstall` once you do that and change this line. It would also be nice if you could start another debian release for the new tag